### DISCUSSION TITLE
How do I enable Askama template autocompletions for a basic Rust project using askama and actix-web?

### DISCUSSION THREAD

**Discussion by ivandimitrov8080** [nix-community/nixvim#777] [Q&A]
I have this template:

```rust
#[derive(Template)]
#[template(path = "user.html")]
struct UserTemplate<'a> {
    name: &'a str,
    text: &'a str,
}
```

and this HTML file:

```html
<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8" />
  <title>Actix Web</title>
</head>

<body>
  <h1>Hi, {{ name }}!</h1>
  <p>
    {{ text }}
  </p>
</body>

</html>
```

There is no autocomplete in the HTML file. I remember some time ago I had autocomplete in there using neovim.

Here's my LSP config:

```nix
          lsp.servers = {
            rust-analyzer = {
              enable = true;
              installCargo = false;
              installRustc = false;
            };
            html.enable = true;
            jsonls.enable = true;
            tailwindcss.enable = true;
            cssls.enable = true;
          };
```

I do have LSP in rust files.

**Comment by traxys** [comment]
What would you want to autocomplete, html tags?

**Reply by ivandimitrov8080** [reply]
The variables `{{ name }}` and `{{ text }}` coming from the rust template.

**Reply by traxys** [reply]
I think you would need an askama aware LSP, but I'm not aware of any.

**Comment by GaetanLepage** [comment]
As this is not a nixvim-related question, I will mark this discussion as "closed" (helps with triage).
Of course, feel free to keep exchanging about this.
